gto/git_utils.py

```diff
@@ -51,6 +52,80 @@ def wrapped_f(*args, **kwargs):
     return wrapped_f
 
 
+def commit_produced_changes_on_auto_commit(
+    message_generator: Callable[..., str] = generate_empty_commit_message
+):
+    """
+    The function `message_generator` can use any argument that the decorated function has.
+
+    Example: here we are using the argument b of the function f to generate the commit message
+
+        def create_message(b: str) -> str:
+            return "commit message with b={b}"
+
+        @commit_produced_changes_on_auto_commit(message_generator=create_message)
+        def f(a: str, b: str, c: str):
+            ...
+
+    """
+
+    def generate_commit_message(**kwargs) -> str:
+        kwargs_for_message_generator = {
+            k: kwargs[k] for k in inspect.getfullargspec(message_generator).args
+        }
+        return message_generator(**kwargs_for_message_generator)
+
+    def wrap(f: Callable):
+        @wraps(f)
+        def wrapped_f(*args, **kwargs):
+            kwargs = _turn_args_into_kwargs(f, args, kwargs)
+
+            if kwargs.get("auto_commit", False) is True:
+                if "repo" in kwargs:
+                    with stashed_changes(
+                        repo_path=kwargs["repo"], include_untracked=True
+                    ) as (stashed_tracked, stashed_untracked):
+                        result = f(**kwargs)
+                        if are_files_in_repo_changed(
+                            repo_path=kwargs["repo"],
+                            files=stashed_tracked + stashed_untracked,
+                        ):
+                            _reset_repo_to_head(repo_path=kwargs["repo"])
+                            raise GTOException(
+                                msg="The command changed files that were not committed, automated committing is not possible. "
+                                "The files have been rolled back to their previous state.\n\n"
+                                "Suggested action: Commit the changes and re-run this command"
```

Comment:
```suggestion
                                msg="The command changed files that were not committed, automated committing is not possible.\n"
                                "Suggested action: Commit the changes and re-run this command"
```
"...have been rolled back..." - user doesn't know we did stash, so no need to tell this to him.

Reply:
What about `The command would have changed files ...` instead of `The command changed files ...`.

I want to make clear to the user that the command did not have any effect and everything is as before.

Reply:
Yes, agreed! Missed that line. Fixed the suggestion above. Looks good now?

Reply:
Changed it in https://github.com/iterative/gto/pull/290/commits/666e7edba0ffaff4b6e10fdb0abb5a0686ff3661